python/src/scipp/__init__.py

```diff
@@ -72,61 +58,15 @@ def custom_formatwarning(msg, *args, **kwargs):
 from ._dataset import *
 from ._groupby import *
 from ._math import *
-from ._operations import *
+from .core.operations import dot, islinspace, issorted, allsorted, cross, sort, \
+        values, variances, stddevs, rebin, where
 from ._unary import *
 from ._reduction import *
 from ._shape import *
 from ._trigonometry import *
-from ._variable import *
-
-setattr(Variable, '_repr_html_', make_html)
-setattr(DataArray, '_repr_html_', make_html)
-setattr(Dataset, '_repr_html_', make_html)
-
-from .io.hdf5 import to_hdf5 as _to_hdf5
-
-setattr(Variable, 'to_hdf5', _to_hdf5)
-setattr(DataArray, 'to_hdf5', _to_hdf5)
-setattr(Dataset, 'to_hdf5', _to_hdf5)
-
-setattr(Variable, 'sizes', property(_make_sizes))
-setattr(DataArray, 'sizes', property(_make_sizes))
-setattr(Dataset, 'sizes', property(_make_sizes))
-
-from ._bins import _bins, _set_bins, _events
-
-setattr(Variable, 'bins', property(_bins, _set_bins))
-setattr(DataArray, 'bins', property(_bins, _set_bins))
-setattr(Dataset, 'bins', property(_bins, _set_bins))
-setattr(Variable, 'events', property(_events))
-setattr(DataArray, 'events', property(_events))
-
-from ._structured import _fields
-
-setattr(
-    Variable, 'fields',
-    property(
-        _fields,
-        doc=
-        """Provides access to fields of structured types such as vectors or matrices."""
-    ))
-
-from ._bins import _groupby_bins
-
-setattr(GroupByDataArray, 'bins', property(_groupby_bins))
-setattr(GroupByDataset, 'bins', property(_groupby_bins))
-
-setattr(Variable, 'plot', plot)
-setattr(DataArray, 'plot', plot)
-setattr(Dataset, 'plot', plot)
-
-# Prevent unwanted conversion to numpy arrays by operations. Properly defining
-# __array_ufunc__ should be possible by converting non-scipp arguments to
-# variables. The most difficult part is probably mapping the ufunc to scipp
-# functions.
-for _cls in (Variable, DataArray, Dataset):
-    setattr(_cls, '__array_ufunc__', None)
-del _cls
+from .core.variable import scalar, zeros, zeros_like, ones, ones_like, empty, \
+        empty_like, full, full_like, matrix, matrices, vector, vectors, array, \
+        linspace, geomspace, logspace, arange
```

Comment:
Are the other `_xxx` modules also supposed to go into the core subpackage?
If not, why is `variable` in there but not `_dataset`?

Reply:
Yes they will, this is WIP.